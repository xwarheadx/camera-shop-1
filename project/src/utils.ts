import { PAGE_LIMIT } from './consts';
import { reviewDataType } from '../src/types/review';

export function getPagesCount(cameraCount: number): number {
  return Math.ceil(cameraCount / PAGE_LIMIT);
}

export function getNumeric(page: string) {
  return Number(page.replace( /^\D+/g, ''));
}

export function seperatePrice(priceValue: number) {
  let returnedPrice = String(priceValue);
  if (returnedPrice.length > 3) {
    returnedPrice = `${returnedPrice.slice(0,2)} ${returnedPrice.slice(2)}`;
  }
  return returnedPrice;
}

export function validateInput(target: (EventTarget & HTMLInputElement) | (EventTarget & HTMLTextAreaElement)) {
  if (target.parentElement?.parentElement && target.name === 'rating' && (Number(target.value) > 0)) {
    if (!target.parentElement.parentElement.parentElement?.className.includes('is-valid')) {
      target.parentElement.parentElement.parentElement?.classList.add('is-valid');
      target.parentElement.parentElement.parentElement?.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name !== 'review' && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length > 0) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string'){
    if(target.value.length === 0) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length >= 5) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name === 'review' && target.value.length < 5) {
    if (!target.parentElement.parentElement.className.includes('is-invalid')) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

}
export function validateForm(data: reviewDataType) {
  let validationStatus = true;
  Object.entries(data).forEach(([key, value]) =>{
    if (key !== 'review') {
      if(value === '') {
        const failedInput = document.querySelector(`input[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
        validationStatus = false;
      }
    }
    if(key === 'review') {
      if(String(value).length < 5) {
        const failedInput = document.querySelector(`textarea[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
      }
    }
    if(key === 'rating') {
      if(value < 1) {
        const failedInput = document.querySelector(`input[name=${key}]`);
        if(failedInput && failedInput.parentElement?.parentElement){
          failedInput.parentElement.parentElement.parentElement?.classList.add('is-invalid');
        }
        validationStatus = false;
      }
    }
  });
  return validationStatus;
}


