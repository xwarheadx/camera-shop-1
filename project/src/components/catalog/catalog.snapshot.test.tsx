import { renderWithProviders } from '../../test/utils/render-with-redux';
import Catalog from './catalog';

test('Render Cataog', () => {
  const { asFragment } = renderWithProviders(<Catalog/>);
  expect(asFragment()).toMatchSnapshot();
});
