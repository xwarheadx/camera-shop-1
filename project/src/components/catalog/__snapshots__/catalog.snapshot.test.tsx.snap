// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Cataog 1`] = `
<DocumentFragment>
  <div
    class="page-content"
  >
    <div
      class="breadcrumbs"
    >
      <div
        class="container"
      >
        <ul
          class="breadcrumbs__list"
        >
          <li
            class="breadcrumbs__item"
          >
            <a
              class="breadcrumbs__link"
              href="/"
            >
              Главная
              <svg
                aria-hidden="true"
                height="8"
                width="5"
              >
                <use
                  xlink:href="#icon-arrow-mini"
                />
              </svg>
            </a>
          </li>
          <li
            class="breadcrumbs__item"
          >
            <span
              class="breadcrumbs__link breadcrumbs__link--active"
            >
              Главная
            </span>
          </li>
        </ul>
      </div>
    </div>
    <section
      class="catalog"
    >
      <div
        class="container"
      >
        <h1
          class="title title--h2"
        >
          Каталог фото- и видеотехники
        </h1>
        <div
          class="page-content__columns"
        >
          <div
            class="catalog__aside"
          >
            <div
              class="catalog-filter"
            >
              <form
                action="#"
              >
                <h2
                  class="visually-hidden"
                >
                  Фильтр
                </h2>
                <fieldset
                  class="catalog-filter__block"
                >
                  <legend
                    class="title title--h5"
                  >
                    Цена, ₽
                  </legend>
                  <div
                    class="catalog-filter__price-range"
                  >
                    <div
                      class="custom-input"
                    >
                      <label>
                        <input
                          name="price"
                          placeholder="от"
                          type="number"
                        />
                      </label>
                    </div>
                    <div
                      class="custom-input"
                    >
                      <label>
                        <input
                          name="priceUp"
                          placeholder="до"
                          type="number"
                        />
                      </label>
                    </div>
                  </div>
                </fieldset>
                <fieldset
                  class="catalog-filter__block"
                >
                  <legend
                    class="title title--h5"
                  >
                    Категория
                  </legend>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        name="photocamera"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Фотокамера
                      </span>
                    </label>
                  </div>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        name="videocamera"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Видеокамера
                      </span>
                    </label>
                  </div>
                </fieldset>
                <fieldset
                  class="catalog-filter__block"
                >
                  <legend
                    class="title title--h5"
                  >
                    Тип камеры
                  </legend>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        name="digital"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Цифровая
                      </span>
                    </label>
                  </div>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        disabled=""
                        name="film"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Плёночная
                      </span>
                    </label>
                  </div>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        name="snapshot"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Моментальная
                      </span>
                    </label>
                  </div>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        disabled=""
                        name="collection"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Коллекционная
                      </span>
                    </label>
                  </div>
                </fieldset>
                <fieldset
                  class="catalog-filter__block"
                >
                  <legend
                    class="title title--h5"
                  >
                    Уровень
                  </legend>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        name="zero"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Нулевой
                      </span>
                    </label>
                  </div>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        name="non-professional"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Любительский
                      </span>
                    </label>
                  </div>
                  <div
                    class="custom-checkbox catalog-filter__item"
                  >
                    <label>
                      <input
                        name="professional"
                        type="checkbox"
                      />
                      <span
                        class="custom-checkbox__icon"
                      />
                      <span
                        class="custom-checkbox__label"
                      >
                        Профессиональный
                      </span>
                    </label>
                  </div>
                </fieldset>
                <button
                  class="btn catalog-filter__reset-btn"
                  type="reset"
                >
                  Сбросить фильтры
                </button>
              </form>
            </div>
          </div>
          <div
            class="catalog__content"
          >
            <div
              class="catalog-sort"
            >
              <form
                action="#"
              >
                <div
                  class="catalog-sort__inner"
                >
                  <p
                    class="title title--h5"
                  >
                    Сортировать:
                  </p>
                  <div
                    class="catalog-sort__type"
                  >
                    <div
                      class="catalog-sort__btn-text"
                    >
                      <input
                        checked=""
                        id="sortPrice"
                        name="sort"
                        type="radio"
                      />
                      <label
                        for="sortPrice"
                      >
                        по цене
                      </label>
                    </div>
                    <div
                      class="catalog-sort__btn-text"
                    >
                      <input
                        id="sortPopular"
                        name="sort"
                        type="radio"
                      />
                      <label
                        for="sortPopular"
                      >
                        по популярности
                      </label>
                    </div>
                  </div>
                  <div
                    class="catalog-sort__order"
                  >
                    <div
                      class="catalog-sort__btn catalog-sort__btn--up"
                    >
                      <input
                        aria-label="По возрастанию"
                        checked=""
                        id="up"
                        name="sort-icon"
                        type="radio"
                      />
                      <label
                        for="up"
                      >
                        <svg
                          aria-hidden="true"
                          height="14"
                          width="16"
                        >
                          <use
                            xlink:href="#icon-sort"
                          />
                        </svg>
                      </label>
                    </div>
                    <div
                      class="catalog-sort__btn catalog-sort__btn--down"
                    >
                      <input
                        aria-label="По убыванию"
                        id="down"
                        name="sort-icon"
                        type="radio"
                      />
                      <label
                        for="down"
                      >
                        <svg
                          aria-hidden="true"
                          height="14"
                          width="16"
                        >
                          <use
                            xlink:href="#icon-sort"
                          />
                        </svg>
                      </label>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              data-testid="loader-test"
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
